pkg/controller/node/node_controller_test.go

```diff
@@ -853,9 +856,21 @@ func TestShouldMakeProgress(t *testing.T) {
 			var oldData, newData, exppatch []byte
 			var err error
 			expNode := nodes[1].DeepCopy()
-			if test.expectTaintsPatch {
-				f.expectGetNodeAction(nodes[1])
+			if test.expectTaintsRemovePatch {
+				var taints []corev1.Taint
+				for _, taint := range expNode.Spec.Taints {
+					if taint.MatchTaint(constants.NodeUpdateInProgressTaint) {
+						continue
+					} else {
+						taints = append(taints, taint)
+					}
+				}
```

Comment:
Possible `// TODO` - dedup this block with the block in `removeUpdateInProgressTaint`